bin/node/src/system_monitor.rs

```diff
@@ -0,0 +1,116 @@
+        Self {
+            sys: System::new_all(),
+            pid: Pid::from(std::process::id() as usize),
+            data_directory,
+        }
+    }
+
+    /// Runs the system monitor loop.
+    pub fn run(&mut self) {
+        loop {
+            std::thread::sleep(MONITOR_INTERVAL);
+            if let Err(err) = self.collect_system_metrics() {
+                error!(target: COMPONENT, ?err, "Error collecting system metrics");
+            }
+        }
+    }
+
+    /// Collects the system metrics.
+    fn collect_system_metrics(&mut self) -> anyhow::Result<()> {
+        self.sys.refresh_all();
+
+        let disks = Disks::new_with_refreshed_list();
+        let system_disk_limit: u64 = disks.iter().map(Disk::total_space).sum();
+        let system_disk_state_available: u64 = disks.iter().map(Disk::available_space).sum();
+        let system_disk_used = system_disk_limit - system_disk_state_available;
+        #[allow(clippy::cast_precision_loss)]
+        let system_disk_utilization = (system_disk_used as f64 / system_disk_limit as f64) * 100.0;
+
+        let system_cpu_utilization = self.sys.global_cpu_usage() / 100.0;
+        let system_cpu_physical_count = System::physical_core_count();
+
+        let system_memory_usage = self.sys.used_memory();
+        let system_memory_limit = self.sys.total_memory();
+        let system_memory_available = self.sys.available_memory();
+        #[allow(clippy::cast_precision_loss)]
+        let system_memory_utilization = system_memory_usage as f64 / system_memory_limit as f64;
+
+        // SAFETY: the process exists since it is the current process.
+        let process = self.sys.process(self.pid).unwrap();
+        let process_memory_usage = process.memory();
+        let process_cpu_utilization = process.cpu_usage() / 100.0;
+        let process_disk_written = process.disk_usage().written_bytes;
+        let process_disk_read = process.disk_usage().read_bytes;
+
+        let (db_file_size, db_wal_size, block_storage_size) = self.collect_store_metrics()?;
+
+        info!(
+            target: COMPONENT,
+            // System memory
+            system_memory_limit,
+            system_memory_available,
+            system_memory_usage,
+            system_memory_utilization,
+            // System CPU
+            system_cpu_physical_count,
+            %system_cpu_utilization,
+            // Disks
+            system_disk_limit,
+            system_disk_state_available,
+            %system_disk_utilization,
+            // Process
+            process_memory_usage,
+            %process_cpu_utilization,
+            process_disk_written,
+            process_disk_read,
+            // Node store data
+            db_file_size,
+            db_wal_size,
+            block_storage_size,
+        );
+
+        Ok(())
+    }
+
+    /// Collects the store metrics.
+    fn collect_store_metrics(&self) -> anyhow::Result<(u64, u64, u64)> {
```

Comment:
nit: Prefer to return a struct rather than a tuple here. Mainly because of the way it would facilitate adding more store-related metrics in the future.